pkg/appsec/config.go

```diff
@@ -0,0 +1,63 @@
+package appsec
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/pkg/errors"
+)
+
+const (
+	// defaultIntakeURLTemplate specifies the string template allowing to obtain
+	// the intake URL from a given site.
+	defaultIntakeURLTemplate = "https://appsecevts-http-intake.logs.%s"
+	// defaultSite is the default intake site.
+	defaultSite = "datadoghq.com"
+)
+
+// Config handles the interpretation of the configuration. It is also a simple
+// structure to share across all the components, with 100% safe and reliable
+// values.
+type Config struct {
+	Enabled bool
+
+	// API configuration group
+	IntakeURL      *url.URL
+	APIKey         string
+	MaxPayloadSize int64
+}
+
+// newConfig creates and returns the AppSec config from the overall agent
+// config.
+func newConfig(cfg config.Config) (*Config, error) {
+	intakeURL, err := intakeURL(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "config")
+	}
+	return &Config{
+		Enabled:        cfg.GetBool("appsec_config.enabled"),
+		IntakeURL:      intakeURL,
+		APIKey:         cfg.GetString("api_key"),
+		MaxPayloadSize: cfg.GetInt64("appsec_config.max_payload_size"),
```

Comment:
Is the value of `0` acceptable? If not, what should it mean? Should it mean "unlimited"? Probably not a good idea. I'd say let's set this only if its `> 0`. WDYT?

Reply:
Yes, it was unlimited if <= 0
I changed it with a default payload size of 5 MB when <= 0